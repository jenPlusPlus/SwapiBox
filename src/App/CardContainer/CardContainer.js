import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';

const CardContainer = ({ cardData, cardType, updateFavorites, favorites }) => {

  const mapped = cardData.map( (item, index) => {

    if (favorites.filter(card => {
      return item.name === card.name;
    }).length > 0) {
      return (<Card cardType={cardData.cardType}
        cardData={cardData[index]}
        key={item.name}
        updateFavorites={updateFavorites}
        isFavorite={'is-favorite'}/>);
    } else {
      return (<Card cardType={cardData.cardType}
        cardData={cardData[index]}
        key={item.name}
        updateFavorites={updateFavorites}
        isFavorite={''}/>);
    }
  });





  if (cardData.length <= 0 && cardType === 'favorites') {
    return (
      <div className="card-container-wrapper">
        No favorite cards found.
      </div>
    );
  } else  {
    return (
      <div className="card-container-wrapper">
        {mapped}
      </div>
    );
  }
};

CardContainer.propTypes = {
  cardData: PropTypes.array,
  cardType: PropTypes.string,
  updateFavorites: PropTypes.func,
  favorites: PropTypes.array
};

export default CardContainer;
