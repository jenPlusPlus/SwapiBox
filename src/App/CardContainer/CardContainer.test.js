import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import fetchMock from 'fetch-mock';
import renderer from 'react-test-renderer';
import { withRouter } from 'react-router';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router, Route } from 'react-router-dom';

describe('CardContainer', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(<CardContainer cardData={[{
    cardType: "people",
    homeworld:"Tatooine",
    homeworldPopulation: "200000",
    name: "Luke Skywalker",
    species:["Human"]
  }]}
    cardType={'people'}
    updatedFavorites={mkFun}
    numFavorites={1}
    favorites={[{
      cardType: "people",
      homeworld:"Tatooine",
      homeworldPopulation: "200000",
      name: "Luke Skywalker",
      species:["Human"]
    }]}
    makeAPICall={mkFun}
                        />);

  it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
