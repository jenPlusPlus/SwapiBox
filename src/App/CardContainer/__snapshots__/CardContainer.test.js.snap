// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardData={
        Array [
            Object {
              "cardType": "people",
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Luke Skywalker",
              "species": Array [
                "Human",
              ],
            },
          ]
    }
    cardType="people"
    favorites={
        Array [
            Object {
              "cardType": "people",
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Luke Skywalker",
              "species": Array [
                "Human",
              ],
            },
          ]
    }
    numFavorites={1}
    updatedFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardData={
                    Object {
                              "cardType": "people",
                              "homeworld": "Tatooine",
                              "homeworldPopulation": "200000",
                              "name": "Luke Skywalker",
                              "species": Array [
                                "Human",
                              ],
                            }
          }
          cardType={undefined}
          isFavorite="is-favorite"
          updateFavorites={undefined}
/>,
      ],
      "className": "card-container-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "cardType": "people",
            "homeworld": "Tatooine",
            "homeworldPopulation": "200000",
            "name": "Luke Skywalker",
            "species": Array [
              "Human",
            ],
          },
          "cardType": undefined,
          "isFavorite": "is-favorite",
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
                        Object {
                                    "cardType": "people",
                                    "homeworld": "Tatooine",
                                    "homeworldPopulation": "200000",
                                    "name": "Luke Skywalker",
                                    "species": Array [
                                      "Human",
                                    ],
                                  }
            }
            cardType={undefined}
            isFavorite="is-favorite"
            updateFavorites={undefined}
/>,
        ],
        "className": "card-container-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "cardType": "people",
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Luke Skywalker",
              "species": Array [
                "Human",
              ],
            },
            "cardType": undefined,
            "isFavorite": "is-favorite",
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
